### PR TITLE
Gui Interactive Controller State Safety

### PR DESCRIPTION
**Pull request by Necktschnagge**

### CODE REVIEW

File: src/gui_interactive_controller.h [hunk 1]
```diff
@@ -48,6 +50,8 @@ struct GameController {
 	using move_path_type = tobor::v1_0::move_path<piece_move_type>;
 
 
+private:
+	friend class GuiInteractiveController; // remove this!!!
```
Comment (Necktschnagge) [inline src/gui_interactive_controller.h:54]: to be resolved in #60

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -208,6 +212,16 @@ void MainWindow::on_actionWEST_triggered()
 
 }
 
+void MainWindow::on_actionForward_triggered()
+{
+	on_actionEAST_triggered(); // change this
```
Comment (Necktschnagge) [inline src/mainwindow.cpp:217]: to be done in #61

File: src/world_generator.h [hunk 3]
```diff
@@ -235,6 +235,32 @@ namespace tobor {
 
 			};
 
+			class initial_state_generator {
```
Comment (Necktschnagge) [inline src/world_generator.h:238]: to be completed in #101

File: src/solver.h [hunk 4]
```diff
@@ -493,6 +493,7 @@ namespace tobor {
 
 			using move_path_type = move_path<piece_move_type>;
 
+			friend class GameController; // remove this! is bad design.
```
Comment (Necktschnagge) [inline src/solver.h:496]: to be solved in #102

**Review by r0ost3r — APPROVED** [review]

**Pull request merged**